File: AtCoder/abc208/B.swift
private func readLn() -> String {
	return readLine() ?? ""
}

private func readInt() -> Int {
	return Int(readLn()) ?? 0
}

var n = readInt()
var coins = [Int]()
var (coin, ans) = (1, 0)
for i in 1 ... 10 {
	coin *= i
	for _ in 1 ... 100 {
		coins.append(coin)
	}
}
for ele in coins.reversed() {
	if ele <= n {
		n -= ele
		ans += 1
	}
}
print(ans)

File: AtCoder/abc208/B.kt
private fun readLn() = readLine() ?: ""
private fun readLns() = readLn().split(" ")
private fun readInt() = readLn().toInt()

fun main() {
    var n = readInt()
    val coins = mutableListOf<Int>()
    var (coin, ans) = listOf(1, 0)
    for (i in 1 until 11) {
        coin *= i
        for (j in 1 until 101) {
            coins.add(coin)
        }
    }
    for (ele in coins.reversed()) {
        if (ele <= n) {
            n -= ele
            ++ans
        }
    }
    print(ans)
}